src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -457,6 +449,97 @@ public DocumentState UpdateAnalyzerConfigSet(ValueSource<AnalyzerConfigSet> newA
                 newTreeSource);
         }
 
+        private ValueSource<TreeAndVersion>? CreateNewTreeSourceForUpdatedAnalyzerConfigSet(ValueSource<AnalyzerConfigSet> newAnalyzerConfigSet)
+        {
+            if (!SupportsSyntaxTree)
+                return null;
+
+            var filePath = GetSyntaxTreeFilePath(this.Attributes);
+
+            // If we don't have an existing tree, there's no optimization we can do.  Just lazily
+            // parse the tree fully when it is requested.
+            if (_treeSource == null)
+                return CreateLazyFullyParsedTree(newAnalyzerConfigSet, filePath);
```

Comment:
Do we want to change this condition to see if the _treeSource exists and has something has been computed for it? My concern being although this avoids a recompute, it's still holding the _treeSource for an item that might still get thrown away eventually.